wavelength-plugins/emoticons.js

```diff
@@ -45,53 +49,69 @@ function parseEmoticons(message) {
 WL.parseEmoticons = parseEmoticons;
 
 exports.commands = {
-	blockemote: 'ignoreemotes',
-	blockemotes: 'ignoreemotes',
-	blockemoticon: 'ignoreemotes',
-	blockemoticons: 'ignoreemotes',
+	blockemote: "ignoreemotes",
+	blockemotes: "ignoreemotes",
+	blockemoticon: "ignoreemotes",
+	blockemoticons: "ignoreemotes",
 	ignoreemotes: function (target, room, user) {
 		this.parse(`/emoticons ignore`);
 	},
 
-	unblockemote: 'unignoreemotes',
-	unblockemotes: 'unignoreemotes',
-	unblockemoticon: 'unignoreemotes',
-	unblockemoticons: 'unignoreemotes',
+	unblockemote: "unignoreemotes",
+	unblockemotes: "unignoreemotes",
+	unblockemoticon: "unignoreemotes",
+	unblockemoticons: "unignoreemotes",
 	unignoreemotes: function (target, room, user) {
 		this.parse(`/emoticons unignore`);
 	},
 
-	emoticons: 'emoticon',
-	emote: 'emoticon',
-	emotes: 'emoticon',
+	emoticons: "emoticon",
+	emote: "emoticon",
+	emotes: "emoticon",
 	emoticon: {
 		add: function (target, room, user) {
-			let parts = target.split(',');
-			for (let u in parts) parts[u] = parts[u].trim();
-			if (!this.can('emote')) return false;
-			if (!parts[1]) return this.sendReply(`Usage: /emoticon add [name], [url] - Remember to resize the image first! (recommended 30x30)`);
-			if (emoticons[parts[0]]) return this.sendReply(`"${parts[0]}" is already an emoticon.`);
-			emoticons[parts[0]] = parts[1];
+			if (!this.can(`emote`)) return false;
+			if (!target) return this.sendReply("Usage: /emoticons add [name], [url]");
```

Comment:
From here until L79:
```js
let [name, url] = target.split(",").map(p => p.trim());
if (!(name && url)) return this.parse(`/help`, true);
```
You should also ensure that the URL is a valid URL and not something random.

Reply:
`this.parse('/help', true)` is used for subcommands to get the help message of the entire group of commands.